src/tools/lifter/bir_lifting_machinesScript.sml

```diff
@@ -746,4 +747,147 @@ val bmr_ms_mem_contains_M0_2 = store_thm ("bmr_ms_mem_contains_M0_2",
+
+val m0_mod_REGS_lifted_imms_LIST_def = Define `
+  m0_mod_REGS_lifted_imms_LIST = [
+    (BMLI (BVar "PSR_C" BType_Bool) (\ms:m0_mod_state. bool2b (ms.base.PSR.C)));
+    (BMLI (BVar "PSR_N" BType_Bool) (\ms:m0_mod_state. bool2b (ms.base.PSR.N)));
+    (BMLI (BVar "PSR_V" BType_Bool) (\ms:m0_mod_state. bool2b (ms.base.PSR.V)));
+    (BMLI (BVar "PSR_Z" BType_Bool) (\ms:m0_mod_state. bool2b (ms.base.PSR.Z)));
+    (BMLI (BVar "R0" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 0w))));
+    (BMLI (BVar "R1" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 1w))));
+    (BMLI (BVar "R2" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 2w))));
+    (BMLI (BVar "R3" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 3w))));
+    (BMLI (BVar "R4" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 4w))));
+    (BMLI (BVar "R5" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 5w))));
+    (BMLI (BVar "R6" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 6w))));
+    (BMLI (BVar "R7" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 7w))));
+    (BMLI (BVar "R8" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 8w))));
+    (BMLI (BVar "R9" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 9w))));
+    (BMLI (BVar "R10" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 10w))));
+    (BMLI (BVar "R11" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 11w))));
+    (BMLI (BVar "R12" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 12w))));
+    (BMLI (BVar "LR" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 14w))));
+    (BMLI (BVar "SP_main" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name F 13w))));
+    (BMLI (BVar "SP_process" (BType_Imm Bit32)) (\ms:m0_mod_state. Imm32 (ms.base.REG (R_name T 13w))));
+    (BMLI (BVar "ModeHandler" BType_Bool) (\ms:m0_mod_state. bool2b (ms.base.CurrentMode = Mode_Handler)));
+    (BMLI (BVar "countw" (BType_Imm Bit64)) (\ms:m0_mod_state. Imm64 (ms.countw)))]`;
+
+
+val m0_mod_REGS_lifted_imms_LIST_REWRS = save_thm ("m0_mod_REGS_lifted_imms_LIST_REWRS",
+  SIMP_RULE (std_ss++wordsLib.WORD_ss) [R_name_def] m0_mod_REGS_lifted_imms_LIST_def);
+
+val m0_mod_lifted_mem_def = Define `
+  m0_mod_lifted_mem = BMLM (BVar "MEM" (BType_Mem Bit32 Bit8)) (\ms:m0_mod_state. ms.base.MEM)`
+
+val m0_mod_lifted_pc_def = Define `
+  m0_mod_lifted_pc = BMLPC (BVar (bir_temp_var_name "PC") (BType_Imm Bit32))
+                       (BVar (bir_temp_var_name "COND") BType_Bool)
+                       (\ms:m0_mod_state. Imm32 (ms.base.REG RName_PC))`
+
+val m0_mod_state_is_OK_def = Define `m0_mod_state_is_OK (ef, sel) (s:m0_mod_state) =
+  ((s.base.AIRCR.ENDIANNESS <=> ef) /\ (s.base.CONTROL.SPSEL <=> sel) /\
+  (s.base.exception = NoException))`
+
+(* Just a dummy for now *)
```

Comment:
Can you elaborate? Explain why it is dummy? I'm sure you will forget about it when you (or someone else) will have to fix that later.

Reply:
that's a copy and modded part. i will have to check, but it's a good point yes.